src/cargo/util/rustc.rs

```diff
@@ -62,4 +75,157 @@ impl Rustc {
             util::process(&self.path)
         }
     }
+
+    pub fn cached_output(&self, cmd: &ProcessBuilder) -> CargoResult<(String, String)> {
+        self.cache.lock().unwrap().cached_output(cmd)
+    }
+}
+
+/// It is a well known that `rustc` is not the fastest compiler in the world.
+/// What is less known is that even `rustc --version --verbose` takes about a
+/// hundred milliseconds! Because we need compiler version info even for no-op
```

Comment:
Could this cc the Cargo/rustc issues as well?

Reply:
Done!